docs/decisions/0005-bulk-enrollment.rst

```diff
@@ -0,0 +1,59 @@
+Bulk Enrollment endpoint for Subscription learners
+--------------------------------------------------
+
+Status
+======
+
+Accepted
+
+Context
+=======
+
+Business need: The admin portal feature to enroll multiple learners in multiple subscription-courses
+needs a suitable backend.
+
+Various enrollment apis exist, but they do not cater to or optimize for multiple enrollments.
+
+There is an `EnrollmentApiClient` which has an `enroll_user_in_course()` method used by:
+  * models::EnterpriseCustomerUser::enroll()
+  * utils::enroll_user() ( handles multiple courses )
+  * views::CourseEnrollmentView::post()
+  * views::HandleConsentEnrollment::get()
+  * views::GrantDataSharingPermission::_enroll_learner_in_course()
+
+There is also a method to create pending user enrollments for non-edX users at:
+  `models::EnterpriseCustomer::enroll_user_pending_registration()`
+This method also handles multiple courses.
+
+Due to the scale and volume considerations, an api endpoint is needed that caters to the bulk
+use case without causing request overload.
```

Comment:
I would add "rate limiting" here as well since that is the technical vocabulary definition of what we ran into.